x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/mapper/SemanticTextFieldMapper.java

```diff
@@ -119,6 +130,11 @@ public NestedObjectMapper getSubMappers() {
         return subMappers;
     }
 
+    @Override
+    public InferenceFieldMetadata getMetadata(Set<String> sourcePaths) {
```

Comment:
Nice - we use the same structure for retrieving metadata directly from the field type